smoke-testing/src/test/java/games/strategy/engine/data/GameSaveTest.java

```diff
@@ -0,0 +1,110 @@
+package games.strategy.engine.data;
+
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+
+import games.strategy.engine.ClientFileSystemHelper;
+import games.strategy.engine.framework.GameDataFileUtils;
+import games.strategy.engine.framework.ServerGame;
+import games.strategy.engine.framework.map.file.system.loader.ZippedMapsExtractor;
+import games.strategy.engine.framework.startup.ui.PlayerTypes;
+import games.strategy.engine.framework.startup.ui.panels.main.game.selector.GameSelectorModel;
+import games.strategy.engine.player.Player;
+import games.strategy.net.LocalNoOpMessenger;
+import games.strategy.net.Messengers;
+import games.strategy.net.websocket.ClientNetworkBridge;
+import games.strategy.triplea.settings.ClientSetting;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+import org.triplea.config.product.ProductVersionReader;
+import org.triplea.game.server.HeadlessLaunchAction;
+import org.triplea.injection.Injections;
+import org.triplea.io.ContentDownloader;
+import org.triplea.io.FileUtils;
+
+/**
+ * Checks that no error is encountered when saving a game on several different maps. This test
+ * downloads the maps in question and starts an all-AI game on each of them, before saving.
+ *
```

Comment:
It is odd that we would be unable to save games that are loaded from a clean XML and latest code. Could you add more detail to this comment about how exactly saving a game can fail?

Reply:
Added more details.